hw3/ThronesAPI/src/CharacterPage.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+import Searchbar from "./Searchbar";
+import { Container, Card } from "react-bootstrap";
+import axios from "axios";
+
+const url = "https://thronesapi.com/api/v2/Characters";
+
+const CharacterPage = () => {
+  const [characters, setCharacters] = useState([]);
+  const [searchResults, setSearchResults] = useState([]);
+
+  useEffect(() => {
+    const fetchCharacters = async () => {
+      try {
+        const response = await axios.get(url);
+        const apiData = response.data;
+        setCharacters(apiData);
+      } catch (error) {
+        console.log("Error: ", error);
+      }
+    };
+    fetchCharacters();
+  }, []);
+
+  return (
+    <div>
+      <Searchbar characters={characters} setSearchResults={setSearchResults} />
+
+      {searchResults.map((character, index) => (
+        <Container style={{ display: "flex", justifyContent: "center" }}>
+          <Card
+            key={index}
+            className="py-3 text-center"
+            style={{ width: "24rem"}}
+          >
```

Comment:
The best way to pick a key is to use a string/number that uniquely identifies a list item among its siblings. Most often you would use IDs from your data as keys. When you don’t have stable IDs for rendered items, you may use the item index as a key as a last resort.

The React docs discourage using indexes for keys, especially if the order of items may change. This can negatively impact performance and may cause issues with component state.